tests/lib/rules/no-shadow-route-definition.js

```diff
@@ -101,6 +101,10 @@ ruleTester.run('no-shadow-route-definition', rule, {
     // With dynamic/variable route or path name:
     'this.route(someVariable);',
     "this.route('views', { path: someVariable })",
+    `this.route(someVariable, function () {
+      this.route('route');
+    });
+    `,
```

Comment:
These should be valid test cases:

```
this.route('first', { path: foo });
this.route('second', { path: 'foo' });
```

```
this.route(foo);
this.route('foo');
```

Reply:
Good catch, obviously current implementation does not support it. I will revise, probably something like we have for params and wildcards, normalising variables into translated value.

Reply:
I added cases 🔝  which meant updating implementation quite a bit to detect which type of argument definition we have for path.

I'll rebase once PR is ready for merging, so it is easier to review diff between commits.